x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/monitor_status/monitor_status_data.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import datemath from '@elastic/datemath';
+import moment from 'moment';
+import {
+  tint,
+  transparentize,
+  VISUALIZATION_COLORS,
+  EuiThemeComputed,
+  EuiThemeColorModeStandard,
+  COLOR_MODES_STANDARD,
+} from '@elastic/eui';
+import type { BrushEvent } from '@elastic/charts';
+import { PingStatus } from '../../../../../../common/runtime_types';
+
+export const SUCCESS_VIZ_COLOR = VISUALIZATION_COLORS[0];
+export const DANGER_VIZ_COLOR = VISUALIZATION_COLORS[VISUALIZATION_COLORS.length - 1];
+export const CHART_CELL_WIDTH = 17;
+
+export interface MonitorStatusTimeBucket {
+  start: number;
+  end: number;
+}
+
+export interface MonitorStatusTimeBin {
+  start: number;
+  end: number;
+  ups: number;
+  downs: number;
+
+  /**
+   * To color code the time bin on chart
+   */
+  value: number;
+}
+
+export interface MonitorStatusPanelProps {
+  /**
+   * Either epoch in millis or Kibana date range e.g. 'now-24h'
+   */
+  from: string | number;
+
+  /**
+   * Either epoch in millis or Kibana date range e.g. 'now'
+   */
+  to: string | number;
+
+  brushable: boolean; // Whether to allow brushing on the chart to allow zooming in on data.
+  periodCaption?: string; // e.g. Last 24 Hours
+  showViewHistoryButton?: boolean;
+  onBrushed?: (timeBounds: { from: number; to: number; fromUtc: string; toUtc: string }) => void;
+}
+
+export function getColorBands(euiTheme: EuiThemeComputed, colorMode: EuiThemeColorModeStandard) {
+  const colorTransitionFn = colorMode === COLOR_MODES_STANDARD.dark ? transparentize : tint;
+
+  return [
+    { color: DANGER_VIZ_COLOR, start: -Infinity, end: -1 },
+    { color: DANGER_VIZ_COLOR, start: -1, end: -0.75 },
+    { color: colorTransitionFn(DANGER_VIZ_COLOR, 0.25), start: -0.75, end: -0.5 },
+    { color: colorTransitionFn(DANGER_VIZ_COLOR, 0.5), start: -0.5, end: -0.25 },
+    { color: colorTransitionFn(DANGER_VIZ_COLOR, 0.75), start: -0.25, end: -0.000000001 },
+    {
+      color: getSkippedVizColor(euiTheme),
+      start: -0.000000001,
+      end: 0.000000001,
+    },
+    { color: colorTransitionFn(SUCCESS_VIZ_COLOR, 0.5), start: 0.000000001, end: 0.25 },
+    { color: colorTransitionFn(SUCCESS_VIZ_COLOR, 0.35), start: 0.25, end: 0.5 },
+    { color: colorTransitionFn(SUCCESS_VIZ_COLOR, 0.2), start: 0.5, end: 0.8 },
+    { color: SUCCESS_VIZ_COLOR, start: 0.8, end: 1 },
+    { color: SUCCESS_VIZ_COLOR, start: 1, end: Infinity },
+  ];
```

Comment:
if i am understanding the purpose correctly , we can explore using 

euiPalettePositive(5); and euiPaletteWarm(5); pallets here